api-inference-community/docker_images/speechbrain/app/pipelines/automatic_speech_recognition.py

```diff
@@ -1,19 +1,57 @@
+from enum import Enum
 from typing import Dict
 
 import numpy as np
 import torch
 from app.pipelines import Pipeline
-from speechbrain.pretrained import EncoderDecoderASR
+from huggingface_hub import HfApi
+from speechbrain.pretrained import EncoderASR, EncoderDecoderASR
+
+
+class ModelType(Enum):
+    ENCODER_ASR = 1
+    ENCODER_DECODER_ASR = 2
+
+
+def interface_to_type(interface_str):
+    if interface_str == "EncoderASR":
+        return ModelType.ENCODER_ASR
+    elif interface_str == "EncoderDecoderASR":
+        return ModelType.ENCODER_DECODER_ASR
+    else:
+        raise ValueError(
+            f"Invalid interface: {interface_str} for Automatic Speech Recognition."
+        )
```

Comment:
seems like this could be reused and not duplicated (`from app.common import interface_to_type`)
With PIpeline validation (I mean supported by ASR vs AudioToAudio) done directly within the `__init__` function.

Reply:
Resolving since this comment is duplicated with the other.